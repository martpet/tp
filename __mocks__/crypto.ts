export default {
  randomBytes: vi.fn().mockReturnValue({
    toString: vi.fn().mockReturnValue('dummyCryptoRandomBytesToString'),
  }),

  createHash: vi.fn().mockReturnValue({
    update: vi.fn().mockReturnValue({
      digest: vi.fn().mockReturnValue('dummyCryptoCreateHashUpdateDigest'),
    }),
  }),
};
