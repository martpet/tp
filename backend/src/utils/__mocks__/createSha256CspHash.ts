export const createSha256CspHash = vi.fn().mockReturnValue('dummySha256CspHash');
