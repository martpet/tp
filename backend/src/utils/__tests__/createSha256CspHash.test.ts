import crypto from 'crypto';

import { createSha256CspHash } from '../createSha256CspHash';

vi.mock('crypto');

const args = ['dummyContent'] as Parameters<typeof createSha256CspHash>;

describe('createSha256CspHash', () => {
  it('calls "createHash" with a correct value', async () => {
    await createSha256CspHash(...args);
    expect(vi.mocked(crypto.createHash).mock.calls).toMatchSnapshot();
  });

  it('calls "update" on the return value of "createHash" with a correct value', async () => {
    await createSha256CspHash(...args);
    expect(vi.mocked(crypto.createHash('').update).mock.calls).toMatchSnapshot();
  });

  it('calls "digest" on the return value of "update" with a correct value', async () => {
    await createSha256CspHash(...args);
    expect(
      vi.mocked(crypto.createHash('').update('').digest).mock.calls
    ).toMatchSnapshot();
  });
});
