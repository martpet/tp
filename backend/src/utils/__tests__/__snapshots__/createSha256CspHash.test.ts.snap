// Vitest Snapshot v1

exports[`createSha256CspHash > calls "createHash" with a correct value 1`] = `
[
  [
    "sha256",
  ],
]
`;

exports[`createSha256CspHash > calls "digest" on the return value of "update" with a correct value 1`] = `
[
  [
    "base64",
  ],
]
`;

exports[`createSha256CspHash > calls "update" on the return value of "createHash" with a correct value 1`] = `
[
  [
    "dummyContent",
  ],
]
`;
